File: gateOne/oldSnacks/Java/CreditCardValidator.java
import java.util.Scanner;

public class CreditCardValidator {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.println("Hello, kindly enter your card details to verify:");
        String cardNumber = input.nextLine();


        String cardType = "";
        if (cardNumber.startsWith("4")) {
            cardType = "Visa Card";
        } else if (cardNumber.startsWith("5")) {
            cardType = "MasterCard";
        } else if (cardNumber.startsWith("37")) {
            cardType = "American Express Card";
        } else if (cardNumber.startsWith("6")) {
            cardType = "Discover Card";
        } else {
            cardType = "Invalid Card";
        }


        int sum1 = 0;
        int sum2 = 0;
        int total = 0;


        boolean doubleDigit = false;

        for (int count = cardNumber.length() - 1; count >= 0; count--) {
            char ch = cardNumber.charAt(count);
            if (!Character.isDigit(ch)) {
                System.out.println("Invalid input. Please enter numeric characters only.");
                return;
            }

            int digit = Character.getNumericValue(ch);

            if (doubleDigit) {
                digit *= 2;
                if (digit > 9) {
                    digit -= 9;
                }
            }
            sum1 += digit;

            doubleDigit = !doubleDigit;
        }

        total = sum1 + sum2;

        String validity = "Invalid";
        if (total % 10 == 0) {
            validity = "Valid";
        }
            System.out.println("\nCredit Card Type: " + cardType);
            System.out.println("Credit Card Number: " + cardNumber);
            System.out.println("Credit Card Digit Length: " + cardNumber.length());
            System.out.println("Credit Card Validity Status: " + validity);
    }
}

File: gateOne/oldSnacks/Python/creditCardValidator.py
def credit_card_validator():
    print("Hello, kindly enter your card details to verify:")
    card_number = input().strip()

    # Determine card type
    if card_number.startswith('4'):
        card_type = "Visa Card"
    elif card_number.startswith('5'):
        card_type = "MasterCard"
    elif card_number.startswith('37'):
        card_type = "American Express Card"
    elif card_number.startswith('6'):
        card_type = "Discover Card"
    else:
        card_type = "Invalid Card"

    # Validate numeric input
    if not card_number.isdigit():
        print("Invalid input. Please enter numeric characters only.")
        return

    sum1 = 0
    double_digit = False

    # Process digits from right to left
    for ch in reversed(card_number):
        digit = int(ch)

        if double_digit:
            digit *= 2
            if digit > 9:
                digit -= 9
        sum1 += digit

        double_digit = not double_digit

    total = sum1
    validity = "Valid" if total % 10 == 0 else "Invalid"

    print("\nCredit Card Type:", card_type)
    print("Credit Card Number:", card_number)
    print("Credit Card Digit Length:", len(card_number))
    print("Credit Card Validity Status:", validity)


credit_card_validator()